### PR TITLE
chore: update privacy tree paper experiment

### PR DESCRIPTION
**Pull request by jfrery**
closes https://github.com/zama-ai/concrete-ml-internal/issues/4089

### CODE REVIEW

**Review by RomanBredehoft — APPROVED** [review]
thanks !

**Review by RomanBredehoft — CHANGES_REQUESTED** [review]
actually, would it be possible to better format the printed tables ? I guess there are meant to be put in a latex format, but it's not very readable here

**Comment by jfrery** [comment]
> actually, would it be possible to better format the printed tables ? I guess there are meant to be put in a latex format, but it's not very readable here

The main point is to copy paste directly to latex. But maybe we can instead print the dataframe yes

**Review by fd0r — DISMISSED** [review]

**Comment by RomanBredehoft** [comment]
> The main point is to copy paste directly to latex. But maybe we can instead print the dataframe yes

I don't really agree, imo the main point of putting this notebook in our directory is to give our users the possibility to be able to see where these timings and plots come from. So it should be kept easily readable like the previous one ! We can keep the latex format  in some internal repo if we want

File: docs/advanced_examples/ExperimentPrivacyTreePaper.ipynb [hunk 1]
```diff
@@ -105,89 +206,114 @@
...
-      "Average precision (fp32): 0.4878108468555631\n",
-      "Average precision: 1.0\n",
-      "Average precision (fp32): 1.0\n",
-      "Average precision: 1.0\n",
-      "Average precision (fp32): 1.0\n",
-      "Average precision: 1.0\n",
-      "Average precision (fp32): 1.0\n"
+      "DecisionTreeClassifier on spambase (#features: 57) -> Acc: 0.9103, Acc (fp32): 0.9035, FHE inference time: 1.59s\n",
+      "XGBClassifier on spambase (#features: 57) -> Acc: 0.9313, Acc (fp32): 0.9362, FHE inference time: 2.01s\n",
+      "RandomForestClassifier on spambase (#features: 57) -> Acc: 0.9089, Acc (fp32): 0.9184, FHE inference time: 6.88s\n",
```
Comment (kcelia) [inline docs/advanced_examples/ExperimentPrivacyTreePaper.ipynb:211]: I agree, this print is not visually appealing.

I can suggest to reformat the print, like:


```
        shift = 52 - len(f"{cls.__name__} on {key_dataset}")

        print(
            f"{cls.__name__} on {key_dataset} {'->':>{shift}} Acc: {np.mean(scores['accuracy']):.4f}, "
            f"Acc (fp32): {np.mean(scores_fp32['accuracy']):.4f}, "
            f"FHE inference time: {scores['inference_time']:.2f}s"
        )

```
image

**Comment by andrei-stoian-zama** [comment]
I think it's not that important that the formatting is nice, it's important that we show our results are reproducible. So @jfrery merge this in the form that you want.

**Review by kcelia — APPROVED** [review]

**Review by RomanBredehoft — APPROVED** [review]

**Pull request merged**